docs/advanced-usage/nesting-test-case-metrics.md

```diff
@@ -93,6 +93,7 @@ class AggregateMetrics(MetricsTestCase):
     macro_F1: float
     mAP: float
     PerClass: List[PerClassMetrics]
+    ThresholdMetrics: Dict[float, PerClassMetrics]
```

Comment:
just want to double check we use float as key is intentional. b/c floating number will have precision issues for instance 
```sh
>>> 0.1+0.2
0.30000000000000004

>>> x = {1.12381902839021830912830981209381092380912830982190380192839081230982190: 42}
>>> x
{1.1238190283902183: 42}
```

Reply:
I've considered the precision issues associated with using floats as keys. However, it's important to note that access to these dictionary values will be managed exclusively by us through the webapp. We plan to control this access via a slider, which will operate on a restricted set of values. This approach mitigates the risks associated with float imprecision.

@munkyshi @yifanwu-kolena Any other use case I am missing where we can have issue?

Reply:
I would consider storing the keys as strings so we can deterministically match them. I don't see any big downside with storing the thresholds in strings and it means we won't have potential precision issues in JS as well once these objects come across the network

Reply:
I would avoid using float as key considering we are using json for data serialization/deserialization. Handling float precision is already something we need to cautious of, and using it as key, which is not valid for json could be more landmines.

Reply:
Side note: is this the right place to update? This PR is about test-sample level metrics, and thresholds for our discussion is scoped to test-sample only (so far), right?